frontend/lib/resolution/Resolver.cpp

```diff
@@ -2161,7 +2161,7 @@ QualifiedType Resolver::typeForId(const ID& id, bool localGenericToUnknown) {
 
   if (asttags::isModule(parentTag)) {
     // If the id is contained within a module, use typeForModuleLevelSymbol.
-    bool isCurrentModule =
+    bool isCurrentModule = !parentId.isEmpty() &&
```

Comment:
An alternative solution might be to set `parentTag` to something "invalid" by default -- we have `chpl::asttags::AST_TAG_UNKNOWN`. Then, the `isModule` check above won't succeed, and `parentTag` will not have a misleading value.

Reply:
I think you're right that we should change that default value, but that doesn't really help with the bug in question. After taking a closer look what I really need to do is create a special case for when the ``id`` is itself a module.